### PR TITLE
Remove DB from internal types

### PR DESCRIPTION
**Pull request by danielmarbach**
As discussed we keep `CosmosDB` into package name, repository name as well as in the namespace but use `Cosmos` only for the types which follows the same reasoning applied to MongoDB persistence

Example

- [Persistence Definition](https://github.com/Particular/NServiceBus.Storage.MongoDB/blob/master/src/NServiceBus.Storage.MongoDB/MongoPersistence.cs#L12)
 - [Namespace](https://github.com/Particular/NServiceBus.Storage.MongoDB/blob/master/src/NServiceBus.Storage.MongoDB/Outbox/MongoOutboxTransaction.cs#L1)

### CODE REVIEW

File: src/NServiceBus.Persistence.CosmosDB/CosmosPersistence.cs [hunk 1]
```diff
@@ -7,9 +7,9 @@
     /// <summary>
     /// CosmosDB Core API persistence
```
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.CosmosDB/CosmosPersistence.cs:8]:
```suggestion
    /// Cosmos DB Core API persistence
```

File: src/NServiceBus.Persistence.CosmosDB/Outbox/OutboxStorage.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@ class OutboxStorage : Feature
 
         protected override void Setup(FeatureConfigurationContext context)
         {
-            var serializer = new JsonSerializer {ContractResolver = new CosmosDBContractResolver()};
+            var serializer = new JsonSerializer {ContractResolver = new UpperCaseIdIntoLowerCaseIdContractResolver()};
```
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.CosmosDB/Outbox/OutboxStorage.cs:22]: 😍

File: src/NServiceBus.Persistence.CosmosDB/SynchronizedStorage/ICosmosStorageSession.cs [hunk 3]
```diff
@@ -5,7 +5,7 @@
     /// <summary>
     ///
```
Comment (SeanFeldman) [inline src/NServiceBus.Persistence.CosmosDB/SynchronizedStorage/ICosmosStorageSession.cs:6]: I'm assuming this will be a separate PR with XML doco updates.

**Review by SeanFeldman — APPROVED** [review]
Approving with one small suggestion that could be handled outside of this PR when XML doco is updated.

File: src/NServiceBus.Persistence.CosmosDB/CosmosPersistence.cs (hunk 1 above)
Comment (danielmarbach) [reply src/NServiceBus.Persistence.CosmosDB/CosmosPersistence.cs:8]: Should be addressed in https://github.com/Particular/NServiceBus.Persistence.CosmosDB/pull/67

File: src/NServiceBus.Persistence.CosmosDB/Outbox/OutboxStorage.cs (hunk 2 above)
Comment (danielmarbach) [reply src/NServiceBus.Persistence.CosmosDB/Outbox/OutboxStorage.cs:22]: Glad you like it 🥇

File: src/NServiceBus.Persistence.CosmosDB/SynchronizedStorage/ICosmosStorageSession.cs (hunk 3 above)
Comment (danielmarbach) [reply src/NServiceBus.Persistence.CosmosDB/SynchronizedStorage/ICosmosStorageSession.cs:6]: See https://github.com/Particular/NServiceBus.Persistence.CosmosDB/pull/67
